megameklab/src/megameklab/printing/PrintInfantry.java

```diff
@@ -99,62 +99,80 @@ protected void writeTextFields() {
+                mp1 = formatMovement(infantry.getJumpMP());
+                mode1 = "Jump";
+                mp2 = formatGroundMP();
+                mode2 = "Ground";
                 break;
             case INF_UMU:
-                setTextField(MP_1, infantry.getActiveUMUCount());
+                mp1 = formatMovement(infantry.getActiveUMUCount());
                 if (infantry.getOriginalJumpMP() > 1) {
-                    setTextField(MODE_1, "SCUBA (Motorized)");
+                    mode1 = "SCUBA (Motorized)";
                 } else {
-                    setTextField(MODE_1, "SCUBA");
+                    mode1 = "SCUBA";
                 }
-                setTextField(MP_2, formatGroundMP(), true);
-                setTextField(MODE_2, "Ground", true);
-                hideElement(MP_2_LABEL, false);
-                hideElement(MODE_2_LABEL, false);
                 break;
             case HOVER:
-                setTextField(MP_1, formatGroundMP());
-                setTextField(MODE_1, "Mechanized Hover");
+                mp1 = formatGroundMP();
+                mode1 = "Mechanized Hover";
                 break;
             case TRACKED:
-                setTextField(MP_1, formatGroundMP());
-                setTextField(MODE_1, "Mechanized Tracked");
+                mp1 = formatGroundMP();
+                mode1 = "Mechanized Tracked";
                 break;
             case WHEELED:
-                setTextField(MP_1, formatGroundMP());
-                setTextField(MODE_1, "Mechanized Wheeled");
+                mp1 = formatGroundMP();
+                mode1 = "Mechanized Wheeled";
                 break;
             case VTOL:
-                setTextField(MP_1, formatMovement(infantry.getJumpMP()));
+                mp1 = formatMovement(infantry.getJumpMP());
                 if (infantry.hasMicrolite()) {
-                    setTextField(MODE_1, "VTOL (Microlite)");
+                    mode1 = "VTOL (Microlite)";
                 } else {
-                    setTextField(MODE_1, "VTOL (Micro-copter)");
+                    mode1 = "VTOL (Micro-copter)";
                 }
                 break;
             case SUBMARINE:
-                setTextField(MP_1, formatMovement(infantry.getActiveUMUCount()));
-                setTextField(MODE_1, "Mechanized SCUBA");
+                mp1 = formatMovement(infantry.getActiveUMUCount());
+                mode1 = "Mechanized SCUBA";
+                // As of writing, the only time Mechanized SCUBA infantry might have nonzero walk mp is if it is beast-mounted.
+                if (infantry.getOriginalWalkMP() > 0) {
+                    mp2 = formatGroundMP();
+                    mode2 = "Ground";
+                }
                 break;
             case INF_MOTORIZED:
-                setTextField(MP_1, formatMovement(infantry.getWalkMP()));
-                setTextField(MODE_1, "Motorized");
+                mp1 = formatMovement(infantry.getWalkMP());
```

Comment:
Is there a reason that this isn't `formatGroundMp` btw? I have no idea so I just left it.